presto-hive/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -304,6 +320,43 @@ private void verifyTableNotExists(String newDatabaseName, String newTableName)
         }
     }
 
+    @Override
+    public synchronized void updateTableStatistics(String databaseName, String tableName, Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        PartitionStatistics originalStatistics = getTableStatistics(databaseName, tableName);
+        PartitionStatistics updatedStatistics = update.apply(originalStatistics);
+
+        Path tableMetadataDirectory = getTableMetadataDirectory(databaseName, tableName);
+        TableMetadata tableMetadata = readSchemaFile("table", tableMetadataDirectory, tableCodec)
+                .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
+
+        TableMetadata updatedMetadata = tableMetadata
+                .withParameters(updateStatisticParameters(tableMetadata.getParameters(), updatedStatistics.getBasicStatistics()))
+                .withColumnStatistics(updatedStatistics.getColumnStatistics());
+
+        writeSchemaFile("table", tableMetadataDirectory, tableCodec, updatedMetadata, true);
+    }
+
+    @Override
+    public synchronized void updatePartitionStatistics(String databaseName, String tableName, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        PartitionStatistics originalStatistics = requireNonNull(
```

Comment:
This is the wrong error message. Should it be `HIVE_PARTITION_DROPPED_DURING_QUERY`? Or a generic `HIVE_METASTORE_ERROR`?

Reply:
I think we should go with the `HIVE_PARTITION_DROPPED_DURING_QUERY` for consistency